frontend/src/page/profile.tsx

```diff
@@ -26,6 +28,44 @@ export const Profile: React.FC = () => {
   }
   return (
     <div>
-        Username: <input type="text" value={user.username}/>
+      <form>
+        <div>
+          <TextField
+            fullWidth
+            label="Username"
+            value={user.username}
+            onChange={(e) => {
+              user.username = e.target.value;
+              setUser(user);
+            }}/>
+        </div>
+        <div>
+          <TextField
+            fullWidth
+            label="Bio"
+            value={user.bio}
+            onChange={(e) => {
+              user.bio = e.target.value;
+              setUser(user);
+            }}/>
+        </div>
+        <Button
+          type="submit"
+          onClick={async (e) => {
+            e.preventDefault();
+            const profileUpdateFetchPromise = abortableJsonFetch<StoredUser>(
```

Comment:
It would be nice if this could be a POST instead. Not sure if we would want to modify `abortableJsonFetch` to allow specifying the HTTP method or create a new method entirely.

Reply:
`abortableJsonFetch` can actually take options as the last parameter, so you can specify `{method: "POST"}`.